apps/archive/archive.py

```diff
@@ -376,18 +376,18 @@ def replace(self, id, document, original):
     def find_one(self, req, **lookup):
         item = super().find_one(req, **lookup)
 
-        if item and str(item.get('task', {}).get('stage', '')) in \
-                get_resource_service('users').get_invisible_stages_ids(get_user().get('_id')):
-            raise SuperdeskApiError.forbiddenError("User does not have permissions to read the item.")
-
         # To fetch the related items json on the basis of ID stored in association for related items
         if item and 'associations' in item:
             for item_name, item_obj in item.get(ASSOCIATIONS).items():
-                if item_obj and 'test' in item_name:
+                if item_obj and self._is_related_content(item_name):
                     item_id = item_obj[config.ID_FIELD]
-                    stored_item = self.find_one(req=None, _id=item_id)
+                    stored_item = super().find_one(req=None, _id=item_id)
```

Comment:
If the item is published a new item is created and this item should be retrieved here.
Also on every update or correction on a published item a new item is created and there is a flag last_published_version that helps you to find the last one.

Reply:
Sorry, my mistake, I think on update you don't have to retrieve the new created item, you still have to find the one with last_published_version set to true.